### PR TITLE
2999 - Access to dental insurance question route (front end)

### PR DESCRIPTION
**Pull request by shawn320**
### Description
Added a new route under intake/id for `Access to dental insurance` question. This pr is for the frontend UI, I will create the api in a separate pr.

### Related Azure Boards Work Items
https://dev.azure.com/DTS-STN/Canada%20Dental%20Care%20Plan/_boards/board/t/Kanban/Stories

### Screenshots (if applicable)
Screenshot 2024-02-29 at 2 23 54 PM

### Checklist

- [ ] I have tested the changes locally
- [ ] I have updated the documentation if necessary
- [ ] I have added/updated tests that prove my fix is effective or that my feature works
- [ ] I have checked that my code follows the project's coding style by running `npm run format:check`
- [ ] I have checked that my code contains no linting errors by running `npm run lint`
- [ ] I have checked that my code contains no type errors by running `npm run typecheck`
- [ ] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [ ] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: frontend/app/components/input-radios.tsx [hunk 1]
```diff
@@ -17,9 +17,10 @@ export interface InputRadiosProps {
   legend: ReactNode;
   name: string;
   required?: boolean;
+  children?: ReactNode;
```
Comment (sebastien-comeau) [inline frontend/app/components/input-radios.tsx:20]: Remove this new prop and use `helpMessagePrimary`. You can use `helpMessagePrimaryClassName` to override the default styling.

File: frontend/app/routes/_public+/intake+/$id+/dental-insurance.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            <InputRadios
+              id="born-type"
+              name="bornType"
+              legend={t('dental-insurance-question:legend')}
+              options={options.map((option) => ({
+                children: option,
+                value: option,
+              }))}
+            >
+              <ul className="mb-4 list-disc pl-4">
```
Comment (sebastien-comeau) [inline frontend/app/routes/_public+/intake+/$id+/dental-insurance.tsx:50]: Use `helpMessagePrimary` prop.

File: frontend/app/routes/_public+/intake+/$id+/dental-insurance.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                  <p>{t('dental-insurance-question:detail.note.content')}</p>
+                </Details>
+                <li>{t('dental-insurance-question:list-item4')}</li>
+              </ul>
+              <p className="mb-4">{t('dental-insurance-question:content2')}</p>
+            </InputRadios>
+          )}
+        </div>
+        <div className="flex flex-wrap items-center gap-3">
+          <ButtonLink id="cancel-button" variant="alternative" to="/personal-information" className="w-24 border-blue-800 py-2 text-blue-800">
```
Comment (sebastien-comeau) [inline frontend/app/routes/_public+/intake+/$id+/dental-insurance.tsx:73]: There's no need to match the mockup buttons at the moment. We can revisit our current button style later on. For now, let's stick with the default variant. If you need a larger button, you can adjust the size using the `size` prop. This also applies to the next button.

**Review by sebastien-comeau — CHANGES_REQUESTED** [review]

File: frontend/app/routes/_public+/intake+/$id+/dental-insurance.tsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                    <ul className="list-disc pl-6">
+                      <li>{t('dental-insurance-question:detail.exception.list-item1')}</li>
+                      <li>{t('dental-insurance-question:detail.exception.list-item2')}</li>
+                    </ul>
+                  </div>
+                </Details>
+                <li>{t('dental-insurance-question:list-item3')}</li>
+                <Details id={t('dental-insurance-question:detail.note.title')} title={t('dental-insurance-question:detail.note.title')}>
+                  <p>{t('dental-insurance-question:detail.note.content')}</p>
+                </Details>
```
Comment (nicholas-ly) [inline frontend/app/routes/_public+/intake+/$id+/dental-insurance.tsx:65]: Placing a `<details>` element as a direct child of an `<ul>` fails HTML validation. Consider placing the `<details>` within the corresponding `<li>` instead.

**Review by nicholas-ly — DISMISSED** [review]

File: frontend/app/routes/_public+/intake+/$id+/dental-insurance.tsx [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            <InputRadios
+              id="born-type"
+              name="bornType"
+              legend={t('dental-insurance-question:legend')}
+              options={options.map((option) => ({
+                children: option,
+                value: option,
+              }))}
+              helpMessagePrimary={helpMessage}
+              helpMessagePrimaryClassName="pl-8 text-[#333333]"
```
Comment (sebastien-comeau) [inline frontend/app/routes/_public+/intake+/$id+/dental-insurance.tsx:78]: Default body color is black.

```suggestion
              helpMessagePrimaryClassName="pl-8 text-black"
```

**Review by sebastien-comeau — CHANGES_REQUESTED** [review]
Final minor tweak.

**Review by sebastien-comeau — DISMISSED** [review]

**Review by sebastien-comeau — APPROVED** [review]

**Pull request merged**